apiserver/facades/client/usermanager/usermanager_test.go

```diff
@@ -821,3 +821,88 @@ func (s *userManagerSuite) TestRemoveUserBulkSharedModels(c *gc.C) {
 	c.Assert(alice.IsDeleted(), jc.IsTrue)
 
 }
+
+func (s *userManagerSuite) TestResetPassword(c *gc.C) {
```

Comment:
While it is at a State level, I would argue that it is part of the contract we are making to users of our API. Hence the statement about putting it as an API test.
There is certainly a valuable debate as to what *should* be tested at what level. Are you *only* testing the code that is between this layer and that, or are you testing the contract of the API.
If we were just testing the code at this layer, then we should be stubbing out State entirely and just asserting that when you call API.Foo() we chain  that into an appropriate call to self.state.FooPla().
I go a bit back and forth, myself. I do like the idea of having tests at the API layer (which is what we expose to other users), of what the precondition/postcondition/side-effects/etc are of interacting with the API. But it does start running into more of an integration test.